Add pgsql status and reset commands to client

RedChops opened the pull request:
This PR adds libpq PQstatus and PQreset functions to the PostgreSQL client.

This is mainly to assist long-lived connections to a database within a connection pool, where a connection may be dropped while the connection is still in the pool due to network or database resets.

RedChops commented:
Actually, let me update the usage of PQReset to PQResetStart so that it is also nonblocking like PQConnectStart in the connect() method. Sorry for the extra commits.

hauptmedia commented:
Hi @RedChops,

would be also nice if you could add a new test or call at least the new functions in the existing test 

https://github.com/openswoole/swoole-src/blob/master/tests/swoole_postgres_coro/query.phpt

Cheers,
Julian

hauptmedia commented:
Btw we should also add the new behaviour for empty results to the test

RedChops commented:
@hauptmedia Definitely can do

RedChops commented:
Alright, I added a couple of tests featuring status() and reset(). Kind of hard to simulate a CONNECTION_BAD event though without external action (i.e. resetting the database or connection in the middle of the test), so I only tested the cases of CONNECTION_OK, both before and after a reset().

I also added a test case for fetchAll() returning an empty array when an empty result set is returned from the database.
